contracts/mainnet/connectors/compound-iii/main.sol

```diff
@@ -0,0 +1,879 @@
+
+		setUint(setId, _amt);
+
+		_eventName = "LogDepositOnBehalf(address,address,address,uint256,uint256,uint256)";
+		_eventParam = abi.encode(market, token, to, _amt, getId, setId);
+	}
+
+	/**
+	 * @dev Deposit base asset or collateral asset supported by the market from 'from' address and update the position of 'to'.
+	 * @notice Deposit a token to Compound for lending / collaterization from a address and update the position of 'to'.
+	 * @param market The address of the market from where to supply.
+	 * @param token The address of the token to be supplied. (For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param from The address from where amount is to be supplied.
+	 * @param to The address on account of which the supply is made or whose positions are updated.
+	 * @param amt The amount of the token to deposit. (For max: `uint256(-1)`)
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of tokens deposited.
+	 */
+	function depositFrom(
+		address market,
+		address token,
+		address from,
+		address to,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	)
+		public
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt = getUint(getId, amt);
+
+		require(
+			market != address(0) && token != address(0),
+			"invalid market/token address"
+		);
+
+		bool isEth = token == ethAddr;
+		address _token = isEth ? wethAddr : token;
+		TokenInterface tokenContract = TokenInterface(_token);
+
+		if (isEth) {
+			convertEthToWeth(isEth, tokenContract, _amt);
+		}
+
+		approve(tokenContract, market, _amt);
+
+		CometInterface(market).supplyFrom(from, to, _token, _amt);
+
+		setUint(setId, _amt);
+
+		_eventName = "LogDepositFrom(address,address,address,address,uint256,uint256,uint256)";
+		_eventParam = abi.encode(market, token, from, to, _amt, getId, setId);
+	}
+
+	/**
+	 * @dev Withdraw base/collateral asset or borrow base asset.
+	 * @notice Withdraw base token or deposited token from Compound.
+	 * @param market The address of the market from where to withdraw.
+	 * @param token The address of the token to be withdrawn. (For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param amt The amount of the token to withdraw. (For max: `uint256(-1)`)
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of tokens withdrawn.
+	 */
+	function withdraw(
+		address market,
+		address token,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	)
+		public
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt = getUint(getId, amt);
+
+		require(
+			market != address(0) && token != address(0),
+			"invalid market/token address"
+		);
+
+		bool isEth = token == ethAddr;
+		address _token = isEth ? wethAddr : token;
+
+		TokenInterface tokenContract = TokenInterface(_token);
+
+		uint256 initialBal = getAccountSupplyBalanceOfAsset(
+			address(this),
+			market,
+			token
```

Comment:
@Richa-iitr We can just check the token balance by calling balanceOf function on tokenContract. no need to call comet contract for this.

Reply:
This would consider for different markets?

Reply:
@Richa-iitr Actually, this method is much cheap than the normal balanceOf function of token